### ISSUE TITLE
"getGroups" not working

### ISSUE THREAD

**Issue by roypardi** [olifolkerd/tabulator#827]
Using version 3.3

I'm calling `getGroups` (on a `groupVisibilityChanged` callback) and only get an empty array returned despite having grouped items. Am I using it wrong?

```
var groups = $("#timeline-states-table").tabulator("getGroups");
console.log('groups',groups);
```

I dug into the Tabulator code (great component, btw!) and see the below getting called. The `dataGrouped` option is false. I tried setting `dataGrouped: true` in the set up options when creating the Tabulator instance but that threw an error.

(FWIW – the functionality I am trying to implement is an exclusive toggle: only one group expanded at a time – am trying to build that off of the `groupVisibilityChanged` callback.)

```
  GroupRows.prototype.getGroups = function () {

      console.log('getGroups called', this.table.options.dataGrouped);
      console.log('this.groupList', this.groupList);
      
      var groupComponents = [];

      if (this.table.options.dataGrouped) {

        this.groupList.forEach(function (group) {

          groupComponents.push(group.getComponent());
        });
      }

      return groupComponents;
    };
```

***Console output***

<img width="281" alt="screen shot 2018-01-15 at 1 12 41 pm" src="https://user-images.githubusercontent.com/9053196/34956147-dcdea71e-f9f5-11e7-8bb3-e3154f3fc4fd.png">

**Comment by roypardi** [comment]
Additional detail: 

So I added the group callback handlers to my tabulator setup and now the `getGroup` function call works. Seems odd. Did I miss this req. in the docs?



```
   dataGrouping:function(){
        console.log('dataGrouping');
    },
    dataGrouped:function(groups){
        console.log('dataGrouped');
    },
```

**Comment by olifolkerd** [comment]
Hey @roypardi 

Thanks for getting in touch,

Could you post a copy of your full table constructor so i can better understand how your table is setup.

Cheers

Oli

**Comment by roypardi** [comment]
This is the gist of it:

```
$("#timeline-states-table").tabulator({
    selectable: "highlight",
    resizableColumns: false,
    movableColumns: false,
    allowReselection: true,
    addRowPos: "bottom",
    movableRows: true,
    height: "90%",
    index: "id",
    layout: "fitColumns",
    groupBy: "groupID",
    groupToggleElement: "arrow",
    // groupToggleElement: "header",
    groupStartOpen: false,
    groupClick: function (e, group) {
        console.log('groupClick');
    },
    groupDblClick: function (e, group) {
        //e - the click event object
        //group - group component
        console.log('groupDblClick');
    },
    groupVisibilityChanged: function (group, visible) {
        manageGroupVisibility(group, visible);
    },
    dataGrouping: function () {
        // console.log('dataGrouping');
    },
    dataGrouped: function (groups) {
        // console.log('dataGrouped');
    },
    groupHeader:function(value, count, data, group){
        return value + "<span style='color:#df0; margin-left:10px;'>(" + count + " item)</span>";
    },

    rowMoved: function (row) {
        states = $("#timeline-states-table").tabulator("getData");
        jsonData.states = states;
        saveProject();
    },
    cellEdited: function (cell) {
        // handle checkox as radio btn

        var obj = cell.getData();

        if (cell.getField() == "initialState") {
            var states = $("#timeline-states-table").tabulator("getData");
            states.forEach(function (element) {
                if (element.id == obj.id) {
                    element.initialState = true;
                } else {
                    element.initialState = false;
                }
            }, this);

            jsonData.states = states;
            $("#timeline-states-table").tabulator('updateData', jsonData.states);
            saveProject();
        } else {
            jsonData.states = $("#timeline-states-table").tabulator("getData");
            saveProject();
            updateStatesMenu();
        }
    },
    columns: [

        { formatter: playIcon, width: 20, align: "center", headerSort: false, editable: false, cellClick: function (e, cell) { stateSelectedHandler(cell) } },
        {
            title: "State", field: "stateName",
            align: "left", headerSort: false, width: 90, editor: "input", editable: editCheck
        },
        {
            title: "First", field: "initialState", width: 30, formatter: "tick", editor: true,
            align: "center", editable: true, headerSort: false, editable: editInitialStateCheck
        },
        { formatter: updateStatesIcon, width: 20, align: "center", headerSort: false, editable: false, cellClick: function (e, cell) { updateStateHandler(cell) } },
        { formatter: removeIcon, width: 20, align: "center", headerSort: false, editable: false, cellClick: function (e, cell) { deleteStateHandler(cell) } },
    ]
});
```

**Comment by olifolkerd** [comment]
Hey @roypardi 

Sorry for the delay in getting back to you on this one. It looks like you found a bug there, It shouldn't have been checking the ***dataGrouped*** option at all.

You will be happy to hear i have pushed a fix for this to the 3.4 branch.

Version 3.4 should be released in the next couple of weeks.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Tabulator version 3.4 has now been officially release!

You can find details of the new release in the [News Section](http://tabulator.info/news)

Full documentation of all the new features can be found [Here](http://tabulator.info/docs/3.4)

Cheers

Oli :)
